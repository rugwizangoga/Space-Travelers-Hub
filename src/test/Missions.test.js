import React from 'react';
import TestRenderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import Missions from '../components/pages/Missions';
import store from '../redux/configureStore';

jest.mock('axios', () => ({
  post: jest.fn(),
  get: jest.fn(),
}));

describe('Missions component', () => {
  it('should render correctly', () => {
    const mission = [
      {
        id: 'mission1',
        mission_name: 'one',
        description: 'a mission',
        joined: true,
      },
    ];

    const MissionsList = TestRenderer.create(
      <Provider store={store}>
        <Missions key={mission.id} MissionsList={mission} />
      </Provider>,
    ).toJSON();
    expect(MissionsList).toMatchSnapshot();
  });
});
