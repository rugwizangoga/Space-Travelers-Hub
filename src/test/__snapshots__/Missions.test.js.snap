// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Missions component should render correctly 1`] = `
<div>
  <table
    className="mission-table"
  >
    <thead>
      <tr>
        <th>
          Mission
        </th>
        <th>
          Description
        </th>
        <th>
          Status
        </th>
        <th>
          Â 
        </th>
      </tr>
    </thead>
    <tbody />
  </table>
</div>
`;
